BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -0,0 +1,116 @@
+	private final WishRepository wishRepository;
+
+	public AccommodationPriceResponse findPrices(String country) {
+		List<Integer> prices = accommodationRepository.findByAccommodationPrices(country);
+		return new AccommodationPriceResponse(calculateAverage(prices), prices);
+	}
+
+	public AccommodationDetailResponse findById(Long id) {
+		Accommodation accommodation = getAccommodation(id);
+		boolean wish = wishRepository.existsByAccommodationId(id);
+		return new AccommodationDetailResponse(accommodation, wish);
+	}
+
+	public AccommodationDetailPriceResponse findDetailPrice(Long id,
+		AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+		Accommodation accommodation = getAccommodation(id);
+		LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+		LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+
+		int date = (int) checkIn.until(checkOut, ChronoUnit.DAYS);
+		int totalPrice = accommodation.getPrice() * date;
+		double discountRate = getDiscountRate(date) * 0.01;
+		int discountPrice = (int) (totalPrice * discountRate);
+		int finalPrice = totalPrice - discountPrice + accommodation.getCleaningFee()
+			+ accommodation.getServiceFee() + accommodation.getAccommodationFee();
+
+		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
+			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
+			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
+	}
+
+	public AccommodationSearchResponse findAccommodations(
+		AccommodationSearchRequest accommodationSearchRequest) {
+		LocalDate checkIn = accommodationSearchRequest.getCheckIn();
+		LocalDate checkOut = accommodationSearchRequest.getCheckOut();
+
+		List<Accommodation> accommodations = accommodationRepository.searchAccommodations(
+			accommodationSearchRequest.getCountry(),
+			LocalDateTime.of(checkIn, LocalTime.of(0, 0)),
+			LocalDateTime.of(checkOut, LocalTime.of(0, 0)),
+			accommodationSearchRequest.getMinPrice(), accommodationSearchRequest.getMaxPrice(),
+			accommodationSearchRequest.getGuestCount());
+
+		List<AccommodationResponse> accommodationResponses = createAccommodationResponses(
+			accommodations, checkIn, checkOut);
+		return new AccommodationSearchResponse(accommodations.size(), accommodationResponses);
+	}
+
+	private List<AccommodationResponse> createAccommodationResponses(
+		List<Accommodation> accommodations, LocalDate checkIn, LocalDate checkOut) {
+		int stayNight = (int) ChronoUnit.DAYS.between(checkIn, checkOut);
+		return accommodations.stream()
+			.map(accommodation -> new AccommodationResponse(accommodation,
+				stayNight * accommodation.getPrice(), accommodation.existsWish()))
+			.collect(Collectors.toList());
+	}
+
+	private Accommodation getAccommodation(Long id) {
+		return accommodationRepository.findById(id)
+			.orElseThrow(() -> new IllegalArgumentException("유효하지 않은 id입니다."));
+	}
```

Comment:
메소드 명에 nullable, throwable 여부를 표현해주는 방식도 있습니다.
사용하는 입장에서 동작을 예측하기 쉬워집니다.
```suggestion
	private Accommodation getAccommodationOrThrow(Long id) {
		return accommodationRepository.findById(id)
			.orElseThrow(() -> new IllegalArgumentException("유효하지 않은 id입니다."));
	}
```

Reply:
몰랐던 부분이었네요..!! 감사합니다 ㅎㅎ